assimilation_code/modules/io/io_filenames_mod.f90

```diff
@@ -654,6 +657,25 @@ subroutine check_attributes(ncFile, filename, ncVarId, domid, varid)
    end select
 endif
 
+if ( get_has_FillValue(   domid, varid) ) then
+   select case (get_xtype(domid, varid))
+      case (NF90_INT)
+         call get_FillValue(domid, varid, spvalINT)
+         call check_attribute_value_int(ncFile, filename, ncVarID, 'FillValue', spvalINT)
+
+      case (NF90_FLOAT)
+         call get_FillValue(domid, varid, spvalR4)
+         call check_attribute_value_r4(ncFile, filename, ncVarID, 'FillValue', spvalR4)
+
+      case (NF90_DOUBLE)
+         call get_FillValue(domid, varid, spvalR8)
+         call check_attribute_value_r8(ncFile, filename, ncVarID, 'FillValue', spvalR8)
+
+      case default
+         call error_handler(E_ERR, 'check_attributes', 'unknown xtype', source)
+   end select
+endif
+
```

Comment:
Now that the variable can have `missing_value` or `_FillValue` we should check them separately.

Reply:
ok, here's where i'd like some explanation of how we are handing having both and if they aren't the same value.

Reply:
The variable can have one or the other, it doesn't have to have both.

Reply:
sure.  technically these 2 attributes mean different things.  missing value is something that's written to say there is no value here.  fill value is supposed to be used for data that hasn't been written yet - not yet filled in.  but people use them interchangeably so it's confusing.  i know they don't have to have both.  i'm asking what happens if they do?  if they're the same value?  if they're different values?  what combination of user choices are we intending to support with this code?